netadapter/netadaptermock/net_adapter_mock.go

```diff
@@ -0,0 +1,182 @@
+package netadaptermock
+
+import (
+	"sync"
+
+	"github.com/kaspanet/kaspad/netadapter/id"
+	"github.com/kaspanet/kaspad/util/mstime"
+
+	"github.com/kaspanet/kaspad/protocol/common"
+
+	"github.com/kaspanet/kaspad/netadapter/router"
+	"github.com/kaspanet/kaspad/wire"
+
+	"github.com/kaspanet/kaspad/config"
+	"github.com/kaspanet/kaspad/netadapter"
+
+	"github.com/pkg/errors"
+)
+
+// NetAdapterMock allows tests and other tools to mockup a simple network adapter without implementing all the required
+// supporting structures.
+type NetAdapterMock struct {
```

Comment:
I don't like using something called "Mock" outside of tests. How about SimpleNetAdapter (after python's SimpleHttpServer)?